trie/transition.go

```diff
@@ -38,6 +38,10 @@ func (t *TransitionTrie) Base() *SecureTrie {
 	return t.base
 }
 
+func (t *TransitionTrie) Overlay() *VerkleTrie {
+	return t.overlay
+}
```

Comment:
Needed for `core/state_processor.go` L173. Double check!

Reply:
is it needed? We could also just implement all the required functions on top of `TransitionTrie`? Not sure which one is the best yet, I'm just not a big fan of exposing too much of the internal structure.

Reply:
I guess it's simpler to do for now, could you just please add a TODO in a comment so that we remember to check that when we merge the final PR ?

Reply:
The "problem" with making it an API in `TransitionTrie` is that the methods are only Verkle related, and not conceptually generic. Still, it might be a good idea.. it's also true that "opening" the box is indirectly giving access to all the APIs.

I'll add the TODO to remember about this, yes.

Reply:
Done.